clippy_lints/src/pass_by_ref_or_value.rs

```diff
@@ -96,37 +127,55 @@ impl<'tcx> TriviallyCopyPassByRef {
                 _ => (),
             }
 
-            if_chain! {
-                if let ty::Ref(input_lt, ty, Mutability::Not) = ty.kind();
-                if !output_lts.contains(&input_lt);
-                if is_copy(cx, ty);
-                if let Some(size) = cx.layout_of(ty).ok().map(|l| l.size.bytes());
-                if size <= self.limit;
-                if let hir::TyKind::Rptr(_, MutTy { ty: ref decl_ty, .. }) = input.kind;
-                then {
-                    let value_type = if is_self_ty(decl_ty) {
-                        "self".into()
-                    } else {
-                        snippet(cx, decl_ty.span, "_").into()
-                    };
-                    span_lint_and_sugg(
-                        cx,
-                        TRIVIALLY_COPY_PASS_BY_REF,
-                        input.span,
-                        &format!("this argument ({} byte) is passed by reference, but would be more efficient if passed by value (limit: {} byte)", size, self.limit),
-                        "consider passing by value instead",
-                        value_type,
-                        Applicability::Unspecified,
-                    );
+            if let ty::Ref(input_lt, ty, Mutability::Not) = ty.kind() {
+                if_chain! {
+                    if !output_lts.contains(&input_lt);
+                    if is_copy(cx, ty);
+                    if let Some(size) = cx.layout_of(ty).ok().map(|l| l.size.bytes());
+                    if size <= self.ref_min_size;
+                    if let hir::TyKind::Rptr(_, MutTy { ty: ref decl_ty, .. }) = input.kind;
+                    then {
+                        let value_type = if is_self_ty(decl_ty) {
+                            "self".into()
+                        } else {
+                            snippet(cx, decl_ty.span, "_").into()
+                        };
+                        span_lint_and_sugg(
+                            cx,
+                            TRIVIALLY_COPY_PASS_BY_REF,
+                            input.span,
+                            &format!("this argument ({} byte) is passed by reference, but would be more efficient if passed by value (limit: {} byte)", size, self.ref_min_size),
+                            "consider passing by value instead",
+                            value_type,
+                            Applicability::Unspecified,
+                        );
+                    }
+                }
+            } else {
+                if_chain! {
+                    if !is_copy(cx, ty) || matches!(ty.kind(), ty::Array(_, _));
```

Comment:
+1 on all of this. I don't think it makes sense for non-`Copy` types, since they are moved and not copied anyway. That means the compiler doesn't need to allocate new space for the moved value.

Reply:
Hum... Aren't values likely copied on move regardless of being `Copy`? I understand you have to "mov" the contents from the stack to registers to do a function call, but I might be wrong. Genuinely curious.

Reply:
Well, I think in the general case space on the stack will be reserved for function parameters, and the value passed as an argument (moved) will be copied into that space.